source/common/http/http3/conn_pool.h

```diff
@@ -53,6 +53,12 @@ class ActiveClient : public MultiplexedActiveClientBase {
     return std::min<int64_t>(quiche_capacity_, effectiveConcurrentStreamLimit());
   }
 
+  // Overridden to return true as long as the client is doing handshake even when it is ready for
+  // early data streams.
```

Comment:
I'd be inclined to name this something more like isHandshakeComplete and do !isHandshakeCompelete with call site comments that as long as the handshake is not complete that it's contributing to connecting stream capacity.  WDYT?

Also I thought we discussed leading this as state == CONNECTED and then overriding for QUIC clients to actually check the handshake status?

/wait

Reply:
Done with renaming. As to the overriding here, it will be a behavior change if I change it to check handshake status as the current call site in h3 pool is checking `connect_timer_ `. So I'm planning to change the condition in the following 0-RTT PR and keep this one a refactoring-only  change.

And unfortunately Http::CodecClient doesn't expose handshake status right now. So some plumbing is needed for that.

Reply:
yeah I still don't like this :-/  How about we dump the connection timer changes, add a boolean, and TODO removing the boolean once we can query the codec client for if the handshake is complete?
/wait

Reply:
done.

Reply:
sorry can we revert the change to where we reset the connection timer?

Reply:
Hmm, I eventually want to move resetting the timer into the if branch because 0-RTT event won't reset the timer.

Reply:
can we save it for the follow up PR, so I can see if there's a better way to do it there?

Reply:
done